File: tests/test_code.py
#!/usr/bin/env python3
"""Test Python file with various issues that our rules should catch."""

# This should trigger python-no-percent-formatting
message = "Hello %s" % name

# This should trigger python-no-bare-except
try:
    risky_operation()
except:
    pass

# This should trigger python-no-print-statements
print("Debug message")

# This should trigger python-no-mutable-defaults
def bad_function(items=[]):
    items.append("new item")
    return items

# This should trigger python-string-comparison
if name is "admin":
    access_granted = True

# This should trigger python-no-deprecated-assert
assert user_valid

# Some good code
def good_function():
    """This function follows best practices."""
    try:
        result = safe_operation()
        return result
    except ValueError as e:
        logger.error(f"Operation failed: {e}")
        raise

File: tests/test_code.js
// Test JavaScript file with various issues that our rules should catch

// This should trigger js-no-var
var oldStyle = "should use let or const";

// This should trigger js-use-strict-equality
if (value == "test") {
    console.log("Should use ===");
}

// This should trigger js-no-console-log
console.log("Debug message");

// This should trigger js-no-function-in-block
if (condition) {
    function badFunction() {
        return "Should not declare functions in blocks";
    }
}

// This should trigger js-no-unused-vars
var unusedVariable = "never used";

// This should trigger js-missing-semicolon
let missingSemicolon = "end without semicolon"

// Some good code
function goodFunction() {
    const message = "This follows best practices";
    return message;
}